### DISCUSSION TITLE
Pass the logger to custom migrations

### DISCUSSION THREAD

**Discussion by RickyRoller** [directus/directus#11247] [Feature Requests]
When writing custom migration scripts it can be hard to debug or report what's going on.

I ended up modifying [run.ts](https://github.com/directus/directus/blob/main/api/src/database/migrations/run.ts#L72) in node_modules in order to pass in the logger. This gave my scripts the ability to write out what was going on in the migration and report any info that might be relevant.

I think it'd be nice to have this as a default for everyone writing custom migrations. ie `await up(database, logger)`.

Open to any thoughts on the subject!

**Comment by jamescammarano** [comment]
I'm no authority on this but I think it would be useful and potentially quick to implement that would really help the dev experience.

I'm very interested to see if anyone has a downside to this.

**Reply by rijkvanzanten** [reply]
Agreed! I think we should make the second parameter a _context_ object, so we can add additional tools in there down the line, like:

```js
async function up(database, { logger }) {
}
```

**Reply by RickyRoller** [reply]
That definitely seems like the way to go. It probably makes sense to add in `log` as well so that the migration can conditionally output more verbose logging.

**Comment by RickyRoller** [comment]
Opened PR https://github.com/directus/directus/pull/11364

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
